packages/hdwallet-walletconnect/src/walletconnect.ts

```diff
@@ -0,0 +1,317 @@
+  }
+
+  public hasOnDevicePassphrase(): boolean {
+    return this.info.hasOnDevicePassphrase();
+  }
+
+  public hasOnDeviceDisplay(): boolean {
+    return this.info.hasOnDeviceDisplay();
+  }
+
+  public hasOnDeviceRecovery(): boolean {
+    return this.info.hasOnDeviceRecovery();
+  }
+
+  public hasNativeShapeShift(srcCoin: core.Coin, dstCoin: core.Coin): boolean {
+    return this.info.hasNativeShapeShift(srcCoin, dstCoin);
+  }
+
+  public supportsOfflineSigning(): boolean {
+    return false;
+  }
+
+  public supportsBroadcast(): boolean {
+    return true;
+  }
+
+  public async clearSession(): Promise<void> {
+    this.disconnect();
+  }
+
+  public async ping(msg: core.Ping): Promise<core.Pong> {
+    // ping function for Wallet Connect? 
+    return { msg: msg.msg };
+  }
+
+  public async sendPin(_pin: string): Promise<void> {
+    // no concept of pin in WalletConnect
+  }
+
+  public async sendPassphrase(_passphrase: string): Promise<void> {
+    // cannot send passphrase. Could show the widget?
+  }
+
+  public async sendCharacter(_charater: string): Promise<void> {
+    // no concept of sendCharacter
+  }
+
+  public async sendWord(_word: string): Promise<void> {
+    // no concept of sendWord
+  }
+
+  public async cancel(): Promise<void> {
+    // no concept of cancel
+  }
+
+  public async wipe(): Promise<void> {
+    return;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  public async reset(_msg: core.ResetDevice): Promise<void> {
+    return;
+  }
+
+  public async recover(_msg: core.RecoverDevice): Promise<void> {
+    // no concept of recover
+  }
+
+  public async loadDevice(_msg: core.LoadDevice): Promise<void> {
+    return;
+  }
+
+  public describePath(msg: core.DescribePath): core.PathDescription {
+    return this.info.describePath(msg);
+  }
+
+  public async getPublicKeys(_msg: Array<core.GetPublicKey>): Promise<Array<core.PublicKey | null>> {
+    // Ethereum public keys are not exposed by the RPC API
+    return [];
+  }
+
+  public async isInitialized(): Promise<boolean> {
+    return true;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  public async disconnect(): Promise<void> {
+    this.provider.disconnect();
+  }
+
+  public async ethSupportsNetwork(chainId = 1): Promise<boolean> {
+    return chainId === 1;
+  }
+
+  public async ethSupportsSecureTransfer(): Promise<boolean> {
+    return false;
+  }
+
+  public ethSupportsNativeShapeShift(): boolean {
+    return false;
+  }
+
+  public async ethSupportsEIP1559(): Promise<boolean> {
+    return false;
+  }
+
+  public ethGetAccountPaths(msg: core.ETHGetAccountPath): Array<core.ETHAccountPath> {
+    return this.info.ethGetAccountPaths(msg);
+  }
+
+  public ethNextAccountPath(msg: core.ETHAccountPath): core.ETHAccountPath | undefined {
+    return this.info.ethNextAccountPath(msg);
+  }
+
+  public async ethGetAddress(): Promise<string | null> {
+    return this.ethAddress;
```

Comment:
```suggestion
    if (!this.ethAddress) {
        if (!this.connected) {
            console.error("WalletConnect provider not connected.");
            return null;
        }
      const { accounts } = this.provider.connector;
      const [address] = accounts;
      this.ethAddress = address;
    }
    return this.ethAddress;
```

Reply:
Is the setting of `ethAddress` inside of it's "getter" an anti-pattern?